includes/object-cache.php

```diff
@@ -699,14 +710,7 @@ public function flush($delay = 0)
             $selective = defined('WP_REDIS_SELECTIVE_FLUSH') ? WP_REDIS_SELECTIVE_FLUSH : null;
 
             if ($salt && $selective) {
-                $script = "
-                    local i = 0
-                    for _,k in ipairs(redis.call('keys', '{$salt}*')) do
-                        redis.call('del', k)
-                        i = i + 1
-                    end
-                    return i
-                ";
+                $script = $this->lua_flush_script($salt);
```

Comment:
Can we split this into two methods, one for selective, non-blocking flush and one for the default block flush when `$unflushable_groups` is empty?

Reply:
I would incorporate this functionality in the function itself.
Regarding the block flush: [using `KEYS` in production environments is discouraged](https://redis.io/commands/KEYS)

How about this flush as substitute if `$unflushable_groups` is empty?
```php
local cur = 0
local i = 0
local tmp
repeat
    tmp = redis.call('SCAN', cur, 'MATCH', '{$salt}*')
    cur = tonumber(tmp[1])
    if tmp[2] then
        for _, v in pairs(tmp[2]) do
            redis.call('del', v)
            i = i + 1
        end
    end
until 0 == cur
return i
```

Reply:
As long as it's blocking, whatever works best. I'd like to avoid possible race conditions with the blocking call.

Reply:
Redis guarantees that an `EVAL` call is atomic so according to the docs [all lua scripts are blocking](https://redis.io/commands/eval#atomicity-of-scripts).

So why use `SCAN` in the first place? According to [this benchmark in the redis repo](https://github.com/antirez/redis/issues/3190#issuecomment-215014512) `SCAN` is faster in lua scripts by a significant factor. Although I could not verify the results on my redis 5 installation it might be an issue for older versions.

Theoretically flushing with `SCAN` will decrease the total amount of database-entries with every page returned. Finding the next page will therefore be generally faster than the last page.

Reply:
I'm easy. If it's faster and backwards compatible, then let's switch 👌🏻